utils.groovy

```diff
@@ -30,12 +30,23 @@ def load_pipecfg() {
     return readYaml(file: "pipecfg/config.yaml")
 }
 
+def substitute_str(str, stream, buildid = "") {
```

Comment:
is groovy like python where you can provider arguments by keyword? so if we had something like:

```
def substitute_str(str, stream = "", buildid = "", basearch = "", foo = "", bar = "", baz = "")
```

Then we could just call:

```
substitute_str("foo=${FOO}", foo="myfoo")
```

and it work without having to pass every possible template token?